dissector/udp.md

```diff
@@ -0,0 +1,285 @@
+```xml
+<P2P>
+  <C2D_C>
+    <uid>95270000YGAKNWKJ</uid>
+    <cli>
+      <port>24862</port>
+    </cli>
+    <cid>849013</cid>
+    <mtu>1350</mtu>
+    <debug>0</debug>
+    <p>MAC</p>
+  </C2D_C>
+</P2P>
+```
+
+**Both**
+
+A camera can do both of the above. In this case it will try to login
+to both udp and tcp. Then the udp will disconnect.
+
+---
+
+## D2C_C_R Payload Camera
+
+Camera replies with this payload on the port specified in `C2D_C`
+
+```xml
+<P2P>
+  <D2C_C_R>
+    <timer>
+      <def>3000</def>
+      <hb>10000</hb>
+      <hbt>60000</hbt>
+    </timer>
+    <rsp>0</rsp>
+    <cid>849013</cid>
+    <did>192</did>
+  </D2C_C_R>
+</P2P>
+```
+
+- **timer** Unknown timer of some sort`
+- **rsp**: Unknown
+- **cid**: The connection ID of the client
+- **did**: The connection ID of the camera
+
+
+
+## T Payload Client
+
+Client then replies with this.
+
+```xml
+<P2P>
+  <C2D_T>
+    <sid>62097899</sid>
+    <conn>local</conn>
+    <cid>82000</cid>
+    <mtu>1350</mtu>
+  </C2D_T>
+</P2P>
+```
+
+- **sid**: ID of the camera
+- **conn**: Type of connection only observed `local` value
+- **cid**: The connection ID of the client
+- **did**: The connection ID of the camera
+- **mtu**: The maximum transmission unit of the connection. Which is the
+           largest packet size in bytes
+
+
+## T Payload Camera
+
+Camera replies with this payload on the port specified in `C2D_C`
+
+**Login**
+
+The client can now login over UDP
+
+
+## Dissconnect
+
+After a message ID 02 (logout) the client send this
+
+```xml
+<P2P>
+  <C2D_DISC>
+    <cid>82000</cid>
+    <did>80</did>
+  </C2D_DISC>
+</P2P>
+```
+
+- **cid**: The connection ID of the client
+- **did**: The connection ID of the camera
+
+The camera also send it but with a `D2C_DISC` tag
+
+## Other
+
+Some cameras seem to also send this before login
+
+```xml
+<P2P>
+  <D2C_CFM>
+    <sid>62097899</sid>
+    <conn>local</conn>
+    <rsp>0</rsp>
+    <cid>82000</cid>
+    <did>80</did>
+    <time_r>0</time_r>
+  </D2C_CFM>
+</P2P>
+```
+
+- **sid**: ID of the camera
+- **conn**: Type of connection only observed `local` value
+- **rsp**: Unknown always 0
+- **cid**: The connection ID of the client
+- **did**: The connection ID of the camera
+- **time_r**: Should be the time but it is always 0
+
+# UDP Ack
+
+These messages are sent to acknowledge receipt of message. They are header only.
+
+## Header 28 Bytes
+
+- 4 Bytes magic: `20cf872a`
+- 4 Bytes Connection ID: This is the connection ID negotiated during UDP Discovery
+- 4 Bytes unknown: Always `00000000` for for UDP Ack
+- 4 Bytes unknown: Always `00000000` for for UDP Ack
+- 4 Bytes Last Packet ID: Last received packet id
+- 4 Bytes Unknown: Observed values `00000000`, `d6010000`, `d7160000` `09e00000` **NEEDS INFO**
+- Payload Size
+- Paload
+
+**To Investigate:**
+  - Why does the unknown byte change. It starts at zero and remains that way
+    for a second. Then seems to change and remain at the new value for another second
```

Comment:
```suggestion
    for a second. Then seems to change and remain at the new value for another second.
    This may be some metric such as average ttl for the last second, or some calculation
    of the jitter
```